### PR TITLE
Fix crash when deterministic MN list is empty and keep paying superblocks in this case

### PR DESCRIPTION
**Pull request by codablock**
This was fixed a while ago in the evo branch and should be on develop as well.

### CODE REVIEW

**Comment by codablock** [comment]
@UdjinM6 I added another commit as I realized that in this (very unlikely) case we'd burn the MN reward from the supply.

**Comment by UdjinM6** [comment]
Hmmm... b9c0a88 has nothing to do with the initial issue fixed by this PR originally. Also, I doubt it fixes anything because I don't see anything broken - we pay miner the whole reward by default https://github.com/dashpay/dash/blob/develop/src/miner.cpp#L176

**Comment by codablock** [comment]
Ahh I see, I also overlooked the loop that follows a few lines below that, it is responsible for subtracting the rewards. Will remove that commit.

However, there is still the very unlikely case that we skip superblock payments in case the MN list is empty. Should I handle this? (by simply not returning)

**Comment by UdjinM6** [comment]
>the very unlikely case that we skip superblock payments

Hmm. That's a good point. Should probably replace `return;` with `LogPrintf("%s -- WARNING: no masternode payments at block %d", __func__, nBlockHeight);` or smth like that. But this PR would need a bit more general title then ;)

File: src/masternode-payments.cpp [hunk 1]
```diff
@@ -264,7 +263,7 @@ void FillBlockPayments(CMutableTransaction& txNew, int nBlockHeight, CAmount blo
         voutMasternodeStr += txout.ToString();
     }
 
-    LogPrint("mnpayments", "%s -- nBlockHeight %d blockReward %lld voutMasternodePaymentsRet \"%s\" txNew %s", __func__,
+    LogPrint("mnpayments", "%s -- nBlockHeight %d blockReward %lld voutMasternodePaymentsRet \"%s\" txNew %s\n", __func__,
```
Comment (UdjinM6) [inline src/masternode-payments.cpp]: This should be reverted - `CTransaction::ToString()` always ends with `\n` by design.

File: src/masternode-payments.cpp [hunk 2]
```diff
@@ -366,7 +365,10 @@ bool CMasternodePayments::GetMasternodeTxOuts(int nBlockHeight, CAmount blockRew
     voutMasternodePaymentsRet.clear();
 
     if(!GetBlockTxOuts(nBlockHeight, blockReward, voutMasternodePaymentsRet)) {
-        assert(!deterministicMNManager->IsDeterministicMNsSporkActive(nBlockHeight));
+        if (deterministicMNManager->IsDeterministicMNsSporkActive(nBlockHeight)) {
+            LogPrintf("CMasternodePayments::%s -- deterministic masternode lists enabled and no payee", __func__);
```
Comment (UdjinM6) [inline src/masternode-payments.cpp]: And this needs `\n` actually.

**Comment by codablock** [comment]
Travis is failing due to crashes in win32 unit tests and usual failures in IX tests. Most likely not related to this PR. Merge? I'll debug issues on develop later today.

**Pull request merged**